File: VB/RuntimeDbChooser.Module.Web/WebCustomAuthentication.vb
Imports System.Web
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Security.Strategy
Imports RuntimeDbChooser.Module.BusinessObjects

Namespace RuntimeDbChooser.Module.Web
    Public Class WebChangeDatabaseAuthenticationActiveDirectory
        Inherits AuthenticationActiveDirectory(Of SecuritySystemUser, CustomLogonParametersForActiveDirectoryAuthentication)

        Public Sub New()
            Me.CreateUserAutomatically = True
        End Sub
        Public Overrides ReadOnly Property IsLogoffEnabled() As Boolean
            Get
                Return True
            End Get
        End Property
        Public Overrides ReadOnly Property AskLogonParametersViaUI() As Boolean
            Get
                Dim databaseName As String = HttpContext.Current.Request.Params(WebChangeDatabaseController.DatabaseParameterName)
                If String.IsNullOrEmpty(databaseName) Then
                    Return True
                End If
                Return MyBase.AskLogonParametersViaUI
            End Get
        End Property
    End Class
End Namespace

File: CS/RuntimeDbChooser.Module.Web/WebCustomAuthentication.cs
using System.Web;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.Security.Strategy;
using RuntimeDbChooser.Module.BusinessObjects;

namespace RuntimeDbChooser.Module.Web {
	public class WebChangeDatabaseAuthenticationActiveDirectory : AuthenticationActiveDirectory<SecuritySystemUser, CustomLogonParametersForActiveDirectoryAuthentication> {
		public WebChangeDatabaseAuthenticationActiveDirectory() {
			this.CreateUserAutomatically = true;
		}
		public override bool IsLogoffEnabled {
			get {
				return true;
			}
		}
		public override bool AskLogonParametersViaUI {
			get {
				string databaseName = HttpContext.Current.Request.Params[WebChangeDatabaseController.DatabaseParameterName];
				if(string.IsNullOrEmpty(databaseName)) {
					return true;
				}
				return base.AskLogonParametersViaUI;
			}
		}
	}
}
